### ISSUE TITLE
//tensorflow/python:saver_test is failing in Windows Bazel build

### ISSUE THREAD

**Issue by meteorcloudy** [tensorflow/tensorflow#12844]
http://ci.tensorflow.org/job/tf-master-win-bzl/1515/console

```
17:24:51 ERROR: testSaveRestore (__main__.LenientNamesTest)
17:24:51 ----------------------------------------------------------------------
17:24:51 Traceback (most recent call last):
17:24:51   File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1321, in _do_call
17:24:51     return fn(*args)
17:24:51   File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1300, in _run_fn
17:24:51     status, run_metadata)
17:24:51   File "C:\Program Files\Anaconda3\lib\contextlib.py", line 66, in __exit__
17:24:51     next(self.gen)
17:24:51   File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\framework\errors_impl.py", line 466, in raise_exception_on_not_ok_status
17:24:51     c_api.TF_GetCode(status))
17:24:51 tensorflow.python.framework.errors_impl.NotFoundError: Key v0 not found in checkpoint
17:24:51 	 [[Node: save_1/RestoreV2 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_arg_save_1/Const_0_0, save_1/RestoreV2/tensor_names, save_1/RestoreV2/shape_and_slices)]]
```
I found out that `TF_SAVER_LENIENT_NAMES` environment variable set at
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/saver_test.py#L2474 is not propagated to the [C++ module](https://github.com/tensorflow/tensorflow/blob/bf6df5e2330dff8383869999840578fa5128e794/tensorflow/core/util/tensor_bundle/tensor_bundle.cc#L597).

The test passes when running with `--test_env=TF_SAVER_LENIENT_NAMES=True`.

This is only happening in the Bazel build, but not in the CMake build. We need to investigate the reason.

**Comment by gunan** [comment]
Looks like the whole build went down again, cannot verify if the issue is really resolved.

**Comment by meteorcloudy** [comment]
@gunan No worries, sent #12900 to fix the build.

**Comment by meteorcloudy** [comment]
@gunan And also, for this issue. I think it's because in Bazel build we use /MT option to statically link into C runtime, but in CMake, we use /MD to dynamically link into C runtime.
That's causing Python code and C++ not sharing the same C runtime in Bazel build, so the environment variable is not propagated from Python to C++.
https://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx

**Comment by gunan** [comment]
Do we set that in TF build files and skylark rules, or does that come from bazel.
I think eventually we want c runtime to be only dynamically linked everywhere.

**Comment by gunan** [comment]
Also, we are now seeing this:
```
00:09:14.146 c:\tmp\_bazel_system\424zmya1\execroot\org_tensorflow\bazel-out\msvc_x64-py3-opt\genfiles\external\local_config_python\numpy_include\numpy\numpyconfig.h(4): fatal error C1083: Cannot open include file: '_numpyconfig.h': No such file or directory
```
Any idea what causes this?

**Comment by snnn** [comment]
Hi @gunan , 
https://github.com/bazelbuild/bazel/issues/2675

And the last few comments in #11735

**Comment by meteorcloudy** [comment]
@gunan Statically linking to C Run Time is currently the default option in Bazel, but it can be reconfigured to dynamic linking easily, I will send a change for this.

For the error, that's a flaky failure caused by https://github.com/bazelbuild/bazel/issues/2675, but unfortunately, we haven't found the root cause for this.

**Comment by meteorcloudy** [comment]
The static linking problem is fixed in Bazel 0.7.0 by https://github.com/bazelbuild/bazel/commit/1edcd0f32d85686c5616a997b988479d3891d525 
and the `Cannot open include file: '_numpyconfig.h': No such file or directory` error will be fixed Bazel 0.8.0 by https://github.com/bazelbuild/bazel/commit/6a4247b10f5cf040c1a7176498bef69c75b1b286.
Closing this issue.
